app/models/notification_setting.rb

```diff
@@ -0,0 +1,34 @@
+# == Schema Information
+#
+# Table name: notification_settings
+#
+#  id          :bigint           not null, primary key
+#  email_flags :integer          default(0), not null
+#  created_at  :datetime         not null
+#  updated_at  :datetime         not null
+#  account_id  :integer
+#  user_id     :integer
+#
+# Indexes
+#
+#  by_account_user  (account_id,user_id) UNIQUE
+#
+
+class NotificationSetting < ApplicationRecord
+  # used for single column multi flags
+  include FlagShihTzu
+
+  belongs_to :account
+  belongs_to :user
+
+  DEFAULT_QUERY_SETTING = {
+    flag_query_mode: :bit_operator
+  }.freeze
+
+  EMAIL_NOTIFCATION_FLAGS = {
+    1 => :new_conversation_when_auto_assign_disabled,
```

Comment:
we can change this flag to `conversation_creation`. based on later discussion with @pranavrajs, We decided not to consider the `round_robin` feature, but avail this notification as a generic option for all conversations created in an inbox that user has access to.

Reply:
So just for the confirmation, this won't take into consideration round-robin at all or will this be triggered only when round-robin is disabled.?

Reply:
@sony-mathew we wouldn't be considering round-robin at all. we will have two generic email notification settings 
- on conversation creation  ( whenever a new conversation is created in an inbox which agent has access to ) 
- on conversation assignment ( when a conversation is assigned to the agent )

Reply:
@sony-mathew This should work independently irrespective of round-robin